ExampleMod/Content/Tiles/ExampleTrap.cs

```diff
@@ -24,51 +26,54 @@ public class ExampleTrap : ModTile
 		}
 
 		// Read the comments above on AddMapEntry.
-		public override ushort GetMapOption(int i, int j) => (ushort)(Main.tile[i, j].frameY / 18);
+		public override ushort GetMapOption(int i, int j) => (ushort)(Main.tile[i, j].TileFrameY / 18);
 
-		public override bool Dangersense(int i, int j, Player player) => true;
+		public override bool IsTileDangerous(int i, int j, Player player) => true;
 
-		public override bool Drop(int i, int j) {
+		// Because this tile does not use a TileObjectData, the correct item drop won't automatically be known, so we must use GetItemDrops to determine it. 
+		public override void GetItemDrops(int i, int j, ref int dropItem, ref int dropItemStack) {
 			Tile t = Main.tile[i, j];
-			int style = t.frameY / 18;
+			int style = t.TileFrameY / 18;
 			// It can be useful to share a single tile with multiple styles.
-			if (style == 0)
-				Item.NewItem(i * 16, j * 16, 16, 16, mod.ItemType(Items.Placeable.ExampleTrap.ExampleTrapA));
-			if (style == 1)
-				Item.NewItem(i * 16, j * 16, 16, 16, mod.ItemType(Items.Placeable.ExampleTrap.ExampleTrapB));
-			return base.Drop(i, j);
+			if (style == 0) {
+				dropItem = Mod.Find<ModItem>(Items.Placeable.ExampleTrap.ExampleTrapA).Type;
+			}
+			if (style == 1) {
+				dropItem = Mod.Find<ModItem>(Items.Placeable.ExampleTrap.ExampleTrapB).Type;
+			}
```

Comment:
I would normally recommend one of:
- `ContentInstance<Items.Placeable.ExampleTrap>.Instances[style].Type`
- A `static ExampleTrap[]` in `ExampleTrap` containing the instances
- A `static GetItem(int style)` function in `ExampleTrap`